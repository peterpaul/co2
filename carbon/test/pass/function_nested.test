/*
 * carbon Compiler and programming language for libco2
 *
 * Copyright (C) 2011,2012 Peterpaul Taekele Klein Haneveld
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
["stdio.h"] int printf(char * format, ...);

int main () {

	int a = 5;

	int doStuff (int b) {
		return a + b;
	}

	printf ("a = 5: doStuff(5) = %d\n", doStuff(5));
	a = 6;
	printf ("a = 6: doStuff(-3) = %d\n", doStuff(-3));
	a = -6;
	printf ("a = -6: doStuff(3) = %d\n", doStuff(3));

	return 0;
}
