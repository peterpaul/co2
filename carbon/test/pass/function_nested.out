a = 5: doStuff(5) = 10
a = 6: doStuff(-3) = 3
a = -6: doStuff(3) = -3
